### PR TITLE
POO-Challenge2

### PR DESCRIPTION
**Pull request by Nels18**

### CODE REVIEW

File: Truck.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    public function isFull()
+    {
+        $sentence = "";
+        if ($this->loading < $this->storageCapacity) {
+            $sentence .= "In filling.";
+        } else {
+            $sentence .= "Full.";
+        }
+        return $sentence;
```
Comment (cedriclombardot) [inline Truck.php]: code style saut de ligne avant return cf https://donkey-school.slack.com/archives/C02E3D6MKS9/p1635929069027400

File: Vehicle.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    public function getCurrentSpeed(): int
+    {
+        return $this->currentSpeed;
+    }
+
+    public function setCurrentSpeed(int $currentSpeed): void
+    {
+        if($currentSpeed >= 0){
```
Comment (cedriclombardot) [inline Vehicle.php]: https://donkey-school.slack.com/archives/C02E3D6MKS9/p1635929069027400

File: Vehicle.php [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    public function brake(): string
+    {
+        $sentence = "";
+        while ($this->currentSpeed > 0) {
+            $this->currentSpeed--;
+            $sentence .= "Brake !!!";
+        }
+
+        $sentence .= "I'm stopped !";
+        return $sentence;
```
Comment (cedriclombardot) [inline Vehicle.php]: https://donkey-school.slack.com/archives/C02E3D6MKS9/p1635929069027400

File: Vehicle.php [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    public function __construct(string $color, int $nbSeats)
+    {
+        $this->color = $color;
+        $this->nbSeats = $nbSeats;
+    }
+
+    public function forward(): string
+    {
+        $this->currentSpeed = 15;
+        return "Go !";
```
Comment (cedriclombardot) [inline Vehicle.php]: https://donkey-school.slack.com/archives/C02E3D6MKS9/p1635929069027400

File: Truck.php [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+    public function getEnergy(): string
+    {
+        return $this->energy;
+    }
+
+    public function setEnergy(string $energy): Car
+    {
+        $this->energy = $energy;
+
```
Comment (cedriclombardot) [inline Truck.php]: https://donkey-school.slack.com/archives/C02E3D6MKS9/p1635929069027400

**Review by Nels18 — COMMENTED** [review]
Corrections faites

**Pull request merged**